packages/react/test/specs/components/Tree/Tree-test.tsx

```diff
@@ -1,7 +1,127 @@
-import { isConformant } from 'test/specs/commonTests'
+import * as React from 'react'
+import * as keyboardKey from 'keyboard-key'
 
+import { isConformant } from 'test/specs/commonTests'
+import { mountWithProvider } from 'test/utils'
 import Tree from 'src/components/Tree/Tree'
+import TreeTitle from 'src/components/Tree/TreeTitle'
+
+const items = [
+  {
+    key: '1',
+    title: '1',
+    items: [
+      {
+        key: '11',
+        title: '11',
+      },
+      {
+        key: '12',
+        title: '12',
+        items: [
+          {
+            key: '121',
+            title: '121',
+          },
+        ],
+      },
+    ],
+  },
+  {
+    key: '2',
+    title: '2',
+    items: [
+      {
+        key: '21',
+        title: '21',
+        items: [
+          {
+            key: '221',
+            title: '221',
+          },
+        ],
+      },
+      {
+        key: '22',
+        title: '22',
+      },
+    ],
+  },
+  {
+    key: '3',
+    title: '3',
+  },
+]
 
 describe('Tree', () => {
   isConformant(Tree)
+
+  describe('activeIndex', () => {
+    it('should contain index of item open at click', () => {
+      const wrapper = mountWithProvider(<Tree items={items} />)
+      const tree = wrapper.find(Tree).at(0)
+
+      wrapper
+        .find(`.${TreeTitle.className}`)
+        .at(0) // title 1
+        .simulate('click')
+      expect(tree.state('activeIndex')).toEqual([0])
```

Comment:
if the component implementation changes (let's say we change name of `activeIndex` state property, reimplement with hooks or use state manager - which will happen soon) then the tests will need to change too;

can you reimplement the expectations just by searching through the DOM elements the open / displayed `TreeItem` elements?

Reply:
refactored the expect